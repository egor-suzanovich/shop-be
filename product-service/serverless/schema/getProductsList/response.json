{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "properties": {
    "body": {
      "type": "array",
      "items": {
        "type": "object",
        "properties": {
          "count": {
            "type": "number"
          },
          "description": {
            "type": "string"
          },
          "id": {
            "type": "string"
          },
          "price": {
            "type": "number"
          },
          "title": {
            "type": "string"
          }
        }
      }
    }
  }
}
