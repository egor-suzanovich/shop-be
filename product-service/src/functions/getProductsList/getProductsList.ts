import { getApiErrorResponse, getApiResponse } from '../../helpers/apiHelpers';
import { getProductsList } from '../../helpers/products';
import commonMidlware from '../../lib/commonMidlware';

export const originalHandler = async (event) => {
  try {
    const result = await getProductsList();
    console.log('[getProductsList handler]', { result });
    return getApiResponse(result);
  } catch (error) {
    console.log(error);
    return getApiErrorResponse();
  }
};

export const handler = commonMidlware(originalHandler);
