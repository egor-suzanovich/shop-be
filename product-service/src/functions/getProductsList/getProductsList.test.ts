import { describe, expect } from '@jest/globals';
import { originalHandler } from './getProductsList';

describe('getProductsList tests', () => {
  it('should return valid response', async () => {
    const response = await originalHandler({});
    expect(response).toMatchSnapshot();
  });
});
