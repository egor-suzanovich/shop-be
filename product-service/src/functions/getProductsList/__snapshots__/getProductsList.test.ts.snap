// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProductsList tests should return valid response 1`] = `
Object {
  "body": "[{\\"count\\":4,\\"description\\":\\"Short Product Description1\\",\\"id\\":\\"7567ec4b-b10c-48c5-9345-fc73c48a80aa\\",\\"price\\":2.4,\\"title\\":\\"S3 Mock Product\\"},{\\"count\\":6,\\"description\\":\\"Short Product Description3\\",\\"id\\":\\"7567ec4b-b10c-48c5-9345-fc73c48a80a0\\",\\"price\\":10,\\"title\\":\\"ProductNew\\"},{\\"count\\":7,\\"description\\":\\"Short Product Description2\\",\\"id\\":\\"7567ec4b-b10c-48c5-9345-fc73c48a80a2\\",\\"price\\":23,\\"title\\":\\"ProductTop\\"},{\\"count\\":12,\\"description\\":\\"Short Product Description7\\",\\"id\\":\\"7567ec4b-b10c-48c5-9345-fc73c48a80a1\\",\\"price\\":15,\\"title\\":\\"ProductTitle\\"},{\\"count\\":7,\\"description\\":\\"Short Product Description2\\",\\"id\\":\\"7567ec4b-b10c-48c5-9345-fc73c48a80a3\\",\\"price\\":23,\\"title\\":\\"Product\\"},{\\"count\\":8,\\"description\\":\\"Short Product Description4\\",\\"id\\":\\"7567ec4b-b10c-48c5-9345-fc73348a80a1\\",\\"price\\":15,\\"title\\":\\"ProductTest\\"},{\\"count\\":2,\\"description\\":\\"Short Product Descriptio1\\",\\"id\\":\\"7567ec4b-b10c-48c5-9445-fc73c48a80a2\\",\\"price\\":23,\\"title\\":\\"Product2\\"},{\\"count\\":3,\\"description\\":\\"Short Product Description7\\",\\"id\\":\\"7567ec4b-b10c-45c5-9345-fc73c48a80a1\\",\\"price\\":15,\\"title\\":\\"ProductName\\"}]",
  "headers": Object {
    "Access-Control-Allow-Headers": "*",
    "Access-Control-Allow-Origin": "*",
    "Cache-Control": "no-cache, no-store, max-age=0",
    "Content-Type": "application/json",
  },
  "statusCode": 200,
}
`;
